### ISSUE TITLE
AAC-ELD in LOAS format

### ISSUE THREAD

**Issue by CoreTechie** [mstorsjo/fdk-aac#87]
AAC profiles LC, HE-AAC and ELD inside LOAS transport formats could be decoded using the "aac-decoder-example" with some modifications. (The modifications included reading the bytes from the encoded bit-stream file only when the decoder internal buffer is empty (after call to aacDecoder_Fill returns valid=0). Else keep feeding the decoder with the aacDecoder_Fill() call in every iteration with the new buffer position and length.)

While playback of LC and HE-AAC decoded wave samples appears to be fine, ELD decoded samples seems to be distorted.

5.1-ELD-LOAS Bitstream:
[Loas.zip](https://github.com/mstorsjo/fdk-aac/files/2170382/Loas.zip)

**Comment by mstorsjo** [comment]
Do similar ELD bistreams decode correctly if using raw transport format (like in m4a)?

**Comment by CoreTechie** [comment]
The same ELD bit-stream in raw format decodes correctly, but the audio is similar to the one in loas format. I used your loas example branch and it also produces the same audio quality (as if it compresses some frequencies)

The ELD Raw format file (.m4a) and the Original wav file from which the bit-streams were encoded can be found in this link:
https://drive.google.com/drive/folders/19ZuBF4LtfoPOKgQkITlZFwwKSaY_EPar 

You can clearly see the difference between the original(uncompressed) and the decoded audio quality.

I understand AAC-ELD compression format was designed mainly for voice / speech communication (mono sound). Does this mean it is not suitable for multi-channel music audio ? Or, some other issue ?

**Comment by mstorsjo** [comment]
I'm not sure if there's something inherently in the ELD profile that makes it impossible to handle multichannel music well, or if it's only a case of encoder tuning and/or bitrate/quality setting. But it's definitely not one of the goals for this profile at least.

You might want to look at opus, to see if that provides better quality for the same latency/bitrate constraints as AAC-ELD, for your usecase.

**Comment by CoreTechie** [comment]
Sure, I would have a look at the Opus codec as well. 

Just an out of context question: 
Can multiple instances of the FDK-AAC decoder be created in a multi-threaded environment, where every instance is handling a different input stream and in a thread safe manner ?

**Comment by mstorsjo** [comment]
Yes, absolutely, it's safe as long as you don't access the same encoder instance from multiple threads at the same time.
